### PR TITLE
DOC: Document more internal testing utilities

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

Documents some additional common test utilities that are meant to be reusable across tests.

---

<details>
<summary>Checklist:</summary>

**Completeness and readability**

- [x] I have updated the documentation to reflect the changes or created a separate PR with updates and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/doc/sources/contribute.rst) for details)_.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] All CI jobs are green or I have provided justification why they aren't.

</details>

### CODE REVIEW

File: doc/sources/contributor-reference.rst [hunk 1]
```diff
@@ -103,13 +103,17 @@ Note that not all estimators offer the same functionalities, and thus tests shou
 
 - ``@pytest.mark.allow_sklearn_fallback``: will avoid having tests fail when they end up calling procedures from |sklearn| instead of from the |onedal|. This can be helpful for example when testing that some corner case falls back correctly when it should.
 - ``onedal.tests.utils._dataframes_support._as_numpy``: this function can be used to convert an input array or data frame to NumPy, regardless of whether it lives on host or on device, and regardless of array API support.
+- ``onedal.tests.utils._dataframes_support.dpnp_available``: a boolean variable that checks for whether the DPNP library is importable at the moment of running the tests. Note that, even if available, it will not be imported automatically in the file where the test happens, so ``import dpnp`` needs to be placed inside the test, or a conditional import needs to be done outside of it.
```
Comment (icfaust) [inline doc/sources/contributor-reference.rst:106]: It would be best to have a general discussion about how data frameworks are not lazily loaded in that file, which are otherwise lazily loaded, because then you'd otherwise have to discuss all frameworks that we may eventually run into (say we increase support of pytorch or dask, or something else.

Comment (david-cortes-intel) [reply doc/sources/contributor-reference.rst:106]: Feel free to add a paragraph.

Comment (icfaust) [reply doc/sources/contributor-reference.rst:106]: sorry, i wont be able to do that until i get back in a month.

**Review by icfaust — APPROVED** [review]
